gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    for item in vec:
+        if type(item) is int or type(item) is long or type(item) is float:
```

Comment:
This check seems unnecessary: `len(item)` will never equal 2 for these types, so the condition below catches all these cases.

Also, as a note, in Python `isinstance` is preferred to `type`.